functions/lib/stripe/create_account.py

```diff
@@ -0,0 +1,103 @@
+#########################
+# STRIPE CREATE ACCOUNT #
+#########################
+
+import json
+import uuid
+from dataclasses import dataclass
+
+import stripe
+from firebase_admin import firestore
+from firebase_functions import https_fn, options
+from google.cloud import firestore
+from google.cloud.firestore import DocumentReference, Transaction
+from lib.constants import db
+from lib.logging import Logger
+from lib.stripe.commons import ERROR_URL
+
+REFRESH_URL = "http://localhost:3000/stripe/refreshAccountLink"
+
+@dataclass
+class CreateStandardStripeAccountRequest:
+  returnUrl: str
+  organiser: str
+  
+  def __post_init__(self):
+    if not isinstance(self.returnUrl, str):
+      raise ValueError("Return Url must be provided as a string.")
+    if not isinstance(self.organiser, str):
+      raise ValueError("Organiser Id must be provided as a string.")
+      
+
+@firestore.transactional
+def check_and_update_organiser_stripe_account(transaction: Transaction, logger: Logger, organiser_ref: DocumentReference, return_url: str, refresh_url: str):
+
+  # Check if organiser exists and attempt to get details
+  maybe_organiser = organiser_ref.get(transaction=transaction)
+  if (not maybe_organiser.exists):
+    logger.error(f"Provided Organiser {organiser_ref.path} was not found in the database.")
+    return json.dumps({"url": ERROR_URL})
+  
+  organiser = maybe_organiser.to_dict()
+
+  # If stripe account id exists and is active, return to previous page
+  if (organiser.get("stripeAccount") != None and organiser.get("stripeAccountActive") == True):
+    logger.info(f"Provided Organiser {organiser_ref.path} already has an active stripe account.")
+    return json.dumps({"url": return_url})
+
+  # 1. first check if the calling organiser already has a stripe account
+  organiser_stripe_account = organiser.get("stripeAccount")
+  if organiser_stripe_account == None:
+    # 2a. if they dont, make a new stripe account and call account link
+    account = stripe.Account.create(type="standard")
```

Comment:
Do we need more parameters during creation?
https://docs.stripe.com/api/accounts/create?lang=python

Reply:
i think we let them do the complete onboarding with stripe, saves us building a workflow ourselves. Alternatively if you think it provides value, we could make this a separate ticket? what do you want?